lib/spack/spack/build_systems/cmake.py

```diff
@@ -103,13 +107,8 @@ def std_cmake_args(self):
     @staticmethod
     def _std_args(pkg):
         """Computes the standard cmake arguments for a generic package"""
-        try:
-            build_type = pkg.build_type()
-        except AttributeError:
-            build_type = 'RelWithDebInfo'
```

Comment:
@ax3l @adamjstewart This method was meant to be used *for a generic package*. Removing the try-except block makes it usable only for `CMakePackage` instances. The problem is that for backward compatibility reasons (not every package migrated to `CMakePackage` and a few of them support different build-systems) we are setting the attribute:
```python
# Standard CMake arguments
m.std_cmake_args = spack.CMakePackage._std_args(pkg)
```
on every package in Spack. Removing the block is very likely the cause of the error that @ax3l noticed in his review.